### PR TITLE
Try to alloc executable memory within rel32 range on Linux machines

### PR DESCRIPTION
**Pull request by maximecb**

### CODE REVIEW

**Comment by maximecb** [comment]
I can confirm that the rel32 range problem is present on both AWS and on the Ubuntu CI machines.

File: yjit_asm.c [hunk 1]
```diff
@@ -124,20 +125,66 @@ x86opnd_t const_ptr_opnd(const void *ptr)
...
-        MAP_PRIVATE | MAP_ANONYMOUS,
-        -1,
-        0
-    );
+    uint8_t* mem_block;
+
+    // On Linux
+    #ifdef MAP_FIXED_NOREPLACE
+        // Align the requested address to page size
+        uint8_t* map_addr = align_ptr((uint8_t*)&alloc_exec_mem, 64 * 1024);
```
[the remark this answers is not in the document]
Comment (maximecb) [reply yjit_asm.c:153]: That might not work on every Linux platform though. Some can have 64KB pages.

**Comment by maximecb** [comment]
Perf seems about the same, maybe slightly improved on AWS:
```
-------------  -----------  ----------  ---------  ----------  -----------  -------
bench          interp (ms)  stddev (%)  yjit (ms)  stddev (%)  interp/yjit  1st itr
30k_ifelse     2400.4       0.0         413.2      0.1         5.81         4.54   
30k_methods    6606.3       0.0         1009.7     0.1         6.54         6.27   
activerecord   173.1        1.4         167.0      0.4         1.04         1.01   
binarytrees    429.5        2.0         383.3      2.5         1.12         1.12   
cfunc_itself   105.1        0.3         61.5       0.9         1.71         1.70   
fannkuchredux  4704.0       0.2         4724.4     0.5         1.00         1.00   
fib            237.3        0.0         77.0       0.2         3.08         3.07   
getivar        105.7        0.3         44.3       0.3         2.39         1.01   
lee            1255.0       0.9         1166.8     1.0         1.08         1.00   
liquid-render  201.3        1.1         178.4      1.2         1.13         1.12   
nbody          125.3        0.1         124.7      0.2         1.00         1.00   
optcarrot      6138.0       1.7         5035.6     0.4         1.22         1.24   
railsbench     3968.0       0.9         3972.7     0.9         1.00         0.98   
setivar        76.1         0.2         34.0       0.3         2.23         1.00   
-------------  -----------  ----------  ---------  ----------  -----------  -------
```

File: yjit_asm.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (XrXr) [reply yjit_asm.c:153]: Maybe call [getpagesize(2)](https://man7.org/linux/man-pages/man2/getpagesize.2.html)?

[the remark this answers is not in the document]
Comment (maximecb) [reply yjit_asm.c]: The manpage says to use sysconf so I'll use that 🤔 

```
       Portable applications should employ sysconf(_SC_PAGESIZE) instead
       of getpagesize():

           #include <unistd.h>
           long sz = sysconf(_SC_PAGESIZE);
```

**Pull request merged**